user-guide/modules/ROOT/pages/getting-started.adoc

```diff
@@ -0,0 +1,886 @@
+When you update your environment variables, you are telling your operating system and other tools where to look for the Boost libraries and headers.
+
+[tabs,sync-group-id=os]
+====
+Windows::
++
+--
+
+[source]
+----
+set BOOST_ROOT=C:\boost <1>
+----
+
+<1> Specify the root directory of the Boost pass:[C++] libraries so other tools can find it
+
+IMPORTANT: Replace `C:\boost` with the prefix directory you specified during the installation, if different.
+
+--
+
+Linux::
++
+--
+
+[source,bash]
+----
+export BOOST_ROOT=/usr/local <1>
+export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH <2>
+export CPLUS_INCLUDE_PATH=/usr/local/include:$CPLUS_INCLUDE_PATH <3>
+----
+
+<1> Specify the root directory of the Boost pass:[C++] libraries so other tools can find it
+<2> Specify additional directories to search for shared libraries when executing a program
+<3> Specify additional directories to search for pass:[C++] header files
+
+IMPORTANT: Replace `/usr/local` with the prefix directory you specified during the installation, if different.
+
+You can add these `export` commands to your `~/.bashrc` or `~/.profile` file to make the changes permanent.
+
+--
+
+macOS::
++
+--
+
+[source,bash]
+----
+export BOOST_ROOT=/usr/local <1>
+----
+
+<1> Specify the root directory of the Boost pass:[C++] libraries so other tools can find it
+
+IMPORTANT: Replace `/usr/local` with the prefix directory you specified during the installation, if different.
+
+You can add these `export` commands to your `~/.bashrc` or `~/.profile` file to make the changes permanent.
+
+--
+====
+
+[#your-first-app]
+== Your First App
+
+=== Header-only libraries
+
+Let’s start by using a header-only library.
+Copy the following program into a file called `example.cpp`.
+
+[source,C++]
+.example.cpp
+----
+#include <boost/lambda/lambda.hpp> <1>
+#include <iostream>
+#include <iterator>
+#include <algorithm>
+
+int main()
+{
+    using namespace boost::lambda;
+    typedef std::istream_iterator<int> in;
+
+    std::for_each(
+        in(std::cin), in(), std::cout << (_1 * 3) << " "); <2>
+}
+----
+
+<1> Including a header from the boost:lambda[] library
+<2> Reads a sequence of integers from standard input, using boost:lambda[] to multiply each number by three and write it to the standard output
+
+We can now compile this program:
+
+[tabs,sync-group-id=build]
+========
+CMake::
++
+--
+You can use https://cmake.org/[CMake,window="_blank"] to configure a project to use Boost and your `example.cpp`.
+Here's an example `CMakeLists.txt` file that uses Boost:
+
+[source,cmake]
+.CMakeLists.txt
+----
+cmake_minimum_required(VERSION 3.0)
+project(MyProject)
+
+find_package(Boost REQUIRED)
+add_executable(MyProject example.cpp)
+target_link_libraries(MyProject Boost::headers)
+----
+
+Store this file next to your `example.cpp` and run the basic CMake workflow:
+
+[source,none,subs="attributes+"]
+----
+mkdir build <1>
+cd build <2>
+cmake .. -D BOOST_ROOT=path/to/{release_basename} <3>
```

Comment:
Does `BOOST_ROOT` work in the general case? I tend to prefer `CMAKE_PREFIX_PATH`, since it's not Boost-specific (you can consume anything by using it).

Reply:
Yes. BOOST_ROOT tends to work even though you don't need it and there are many other ways. But setting the environment just gets it for free.